### PR TITLE
initial rxjs-exec type definition

### PR DESCRIPTION
**Pull request by mabels**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [n] The package does not provide its own types, and you can not add them.
- [y]  If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [n] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [n] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by sheetalkamat** [comment]
@mabels Please fix the build break

File: package.json [hunk 1]
```diff
@@ -21,10 +21,11 @@
...
   "devDependencies": {
-    "dtslint": "Microsoft/dtslint#production",
+    "dtslint": "github:Microsoft/dtslint#production",
     "types-publisher": "Microsoft/types-publisher#production"
   },
   "dependencies": {
-    "jslint": "^0.10.3"
+    "jslint": "^0.10.3",
+    "typescript": "^2.4.2"
   }
```
Comment (sheetalkamat) [inline package.json]: Whats the need for this change? adding @andy-ms since he knows more about this than myself.

File: types/rxjs-exec/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import * as Rx from 'rxjs';
+
+export class Options {
+    binary?: boolean;
+    input?: string|Buffer;
+    stdin?: NodeJS.WritableStream;
```
Comment (sheetalkamat) [inline types/rxjs-exec/index.d.ts]: stdin passed needs to be Boolean ? https://github.com/kssfilo/rxjs-exec#connecting-processstdin

File: types/rxjs-exec/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+export function exec(commandLine: string, options?: Options): Rx.Observable<Buffer>;
+export function execFilter(commandLine: string, options?: Options): Rx.Observable<Buffer>;
+export function mapExec(commandLine: string, options?: Options): Rx.Observable<Buffer>;
+export function mapExecFilter(commandLine: string, options?: Options): Rx.Observable<Buffer>;
+// declare export patch(rx: Rx.Observer);
```
Comment (sheetalkamat) [inline types/rxjs-exec/index.d.ts]: why is patch method missing ? https://github.com/kssfilo/rxjs-exec#install

File: types/rxjs-exec/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// Definitions by: Meno Abels <https://github.com/mabels>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import * as Rx from 'rxjs';
+
+export class Options {
+    binary?: boolean;
+    input?: string|Buffer;
```
Comment (sheetalkamat) [inline types/rxjs-exec/index.d.ts]: the documentation doesn't talk about input option. https://www.npmjs.com/package/rxjs-exec are you sure this option is present and correctly typed?

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/rxjs-exec/index.d.ts (hunk 4 above)
Comment (mabels) [inline types/rxjs-exec/index.d.ts]: i read the source instead of the documentation and the source has input.
   ref = options != null ? options : {}, input = ref.input, stdin = ref.stdin, binary = ref.binary;

File: types/rxjs-exec/index.d.ts (hunk 2 above)
Comment (mabels) [inline types/rxjs-exec/index.d.ts]: sorry that was my fault, you are absolutely right.
I modified my PR.

File: types/rxjs-exec/index.d.ts (hunk 3 above)
Comment (mabels) [inline types/rxjs-exec/index.d.ts]: Hi,

i think the patch method is not ok for typescript, it monkey-patch rxjs module.
So you could apply this patch during runtime but rxjs will still not have the types.
So i decided to remove the patch function in the type definition.

File: package.json (hunk 1 above)
Comment (andy-ms) [inline package.json]: Any changes to the global `package.json` should go in their own PRs. I created #18746 to undo the addition of `jslint`.

File: package.json [hunk 5]
```diff
@@ -21,7 +21,7 @@
     "lint": "dtslint types"
   },
   "devDependencies": {
-    "dtslint": "Microsoft/dtslint#production",
+    "dtslint": "github:Microsoft/dtslint#production",
```
Comment (minestarks) [inline package.json]: Why is this needed?

File: types/rxjs-exec/index.d.ts (hunk 3 above)
Comment (minestarks) [reply types/rxjs-exec/index.d.ts]: I'm not sure what you mean.  `require('rxjs-exec').patch(Rx.Observable);` is valid code and should be compilable right? If so, it would make sense to include it in the typing.

If `patch` doesn't belong here, then please remove the commented out code.

I can merge once you change hte code either way.

**Review by minestarks — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (andy-ms) [reply package.json]: I mentioned above, any changes to the global `package.json` should go in their own PRs.

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has unaddressed reviewer comments and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.
